libs/financial-aid/shared/src/lib/formatters.ts

```diff
@@ -194,12 +200,79 @@ export const getFileTypeName: KeyMapping<FileType, string> = {
   SpouseFiles: 'Gögn frá maka',
 }
 
-export const getEmailTextFromState: KeyMapping<ApplicationState, string> = {
-  New: 'Umsókn þín er móttekin',
-  DataNeeded: 'Okkur vantar gögn til að klára að vinna úr umsókninni',
-  InProgress: 'Umsókn þín er móttekin og er nú í vinnslu',
-  Rejected: 'Umsókn þinni um aðstoð hefur verið synjað',
-  Approved: 'Umsóknin þín er samþykkt og áætlun er tilbúin',
+export const getApplicantEmailDataFromState = (
+  state:
+    | ApplicationEventType.NEW
+    | ApplicationEventType.DATANEEDED
+    | ApplicationEventType.REJECTED
+    | ApplicationEventType.APPROVED,
+  linkToStatusPage: string,
+  applicantEmail: string,
+  municipality: Municipality,
+  typeOfDataNeeded?: string,
+): { subject: string; data: ApplicantEmailData } => {
+  switch (state) {
+    case ApplicationEventType.NEW:
+      return {
+        subject: 'Umsókn fyrir fjárhagsaðstoð móttekin',
+        data: {
+          header: `Umsókn þín fyrir ${months[nextMonth]} er móttekin og er nú í vinnslu`,
+          content:
+            'Umsóknin verður afgreidd eins fljótt og auðið er. Þú færð annan tölvupóst þegar vinnsla klárast eða ef við þurfum einhver gögn beint frá þér.<br><br>Þú getur fylgst með stöðu umsóknar, sent inn spurningar, o.fl. í þeim dúr á stöðusíðu umsóknarinnar. Kíktu á hana með því að smella á hnappinn fyrir neðan.',
+          applicationChange: 'Umsókn móttekin og í vinnslu',
+          applicationMonth: months[nextMonth],
+          applicationYear: getNextPeriod.year,
+          statusPageUrl: linkToStatusPage,
+          applicantEmail,
+          municipality,
+        },
+      }
+
+    case ApplicationEventType.DATANEEDED:
+      return {
+        subject: 'Okkur vantar gögn til að klára að vinna úr umsókninni',
+        data: {
+          header: `Okkur vantar gögn til að klára að vinna úr umsókninni`,
+          content: `Við þurfum að sjá <strong>${typeOfDataNeeded}</strong>. Smelltu á hnappinn til að heimsækja þína stöðusíðu þar sem þú getur sent okkur gögn.`,
+          applicationChange: 'Umsóknin bíður eftir gögnum',
+          applicationMonth: months[nextMonth],
+          applicationYear: getNextPeriod.year,
+          statusPageUrl: linkToStatusPage,
+          applicantEmail,
+          municipality,
+        },
+      }
+
+    case ApplicationEventType.REJECTED:
+      return {
+        subject: 'Umsókn þinni um aðstoð hefur verið synjað',
+        data: {
+          header: 'Umsókn þinni um aðstoð hefur verið synjað',
+          content: `Umsókn þinni um fjárhagsaðstoð í ${months[nextMonth]} hefur verið synjað á grundvelli 12. gr.: Tekjur og eignir umsækjanda. Smelltu á hlekkinn hér fyrir neðan til að kynna þér reglur um fjárhagsaðstoð.`,
```

Comment:
grundvelli 12. gr.: Tekjur og eignir umsækjanda. are we sure we want to have this hardcoded? Can it be fore some other reasons?

Reply:
Good question, I think this is the general article about why you could reject financial aid and in that article the reasons are counted.

That being said I did not look it up 🙈 

But judged it by what is said to the employee who rejects the application which is:
`Þegar umsóknum er synjað er hlekkur á slóð um reglur um fjárhagsaðstoð sveitarfélagsins birtur í tölvupósti sem er sendur á umsækjanda. Í ástæðu synjunar er því óhætt að vísa til reglna um fjárhagsaðstoð.`